plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -355,4 +378,18 @@ public String projectsRoot() {
 
         return serversConf;
     }
+
+    /**
+     * Checks environment variable {@code CHE_DOCKER_USE_INTERNAL_CONTAINER_ADDRESS} to determine if
+     * direct addresses of containers should be used.
+     * @return true if {@code CHE_DOCKER_USE_INTERNAL_CONTAINER_ADDRESS} is "true", false otherwise.
+     */
+    protected boolean useInternalContainerAddresses() {
+        String useInternalContainerAddresses = System.getenv(CHE_DOCKER_USE_INTERNAL_CONTAINER_ADDRESS);
```

Comment:
Sorry, could you clarify this point? Do you mean that it is preferable to control this behaviour through a property defined in `che.properties`?